futures-test/src/task/context.rs

```diff
@@ -0,0 +1,146 @@
+use crate::task::{executor, wake};
+use futures_core::{
+    future::Future,
+    task::{local_waker, Context, Executor, Poll, Wake},
```

Comment:
- For consistency with the rest of our codebase the non-nested style would be better
- `Context` should not be imported directly. Instead it should always be referred to as `task::Context`. This decision was made in the 0.2 days. I wasn't around back then, but I think it's because the word "context" is so nondescript.

Reply:
I think in this case it's better to use a bare `Context`, this is explicitly a helper for managing that exact type. I think I'll change what's used in the docs though.

Reply:
In terms of non-nested style, what about

```rust
use crate::task::{executor, wake};
use std::{marker::Unpin, mem::PinMut, sync::Arc};
```

I assume that the first is definitely ok, but what about the second? What's the actual limit on nesting that you want to enforce?

Reply:
I'd prefer to expand the second one as well. I'd like to align our code style with the code style of the standard library and compiler. I browsed the code just now and it seems that they consistently expand such things. Also, our existing code does so as well:

```rust
use core::marker::Unpin;
use core::mem::PinMut;
```
can be found many many times in futures-rs 🙂

Reply:
Alright, so nesting the final segment in a path is ok, but no more than that.